src/proof/nova.rs

```diff
@@ -322,6 +322,46 @@ mod tests {
         }
     }
 
+    fn check_emitted_frames<Fo: Fn(&'_ mut Store<Fr>) -> Vec<Ptr<Fr>>>(
+        source: &str,
+        chunk_frame_count: usize,
+        limit: usize,
+        emitted: Fo,
+    ) {
+        let mut s = Store::default();
+
+        let expr = s.read(source).unwrap();
+
+        let e = empty_sym_env(&s);
+
+        let nova_prover = NovaProver::<Fr>::new(chunk_frame_count);
+        let proof_results = Some(
+            nova_prover
+                .evaluate_and_prove(expr, empty_sym_env(&s), &mut s, limit)
+                .unwrap(),
+        );
+
+        let shape_and_gens = nova_prover.make_shape_and_gens();
+
+        if let Some((proof, instance)) = proof_results {
+            proof.verify(&shape_and_gens, &instance);
+        }
```

Comment:
This seems to just be generating a Nova proof and checking it, which is unnecessary for the frame-checking. Especially if this is used as a supplement to existing tests, it may be superfluous. We should have the option of avoiding this expensive check (or always skipping — since we're not even checking the proof of the emitted values in particular).

We do need to check the constraint systems for each frame that emits a value, but that should be included in the checks that already exist elsewhere (assuming this function is called on expressions whose evaluation is already being tested). In that case, it's probably sufficient to verify that the expected emitted values do indeed correspond to the intermediate frames for which circuits were synthesized.